feat : 포만감 10 초과, 포만감 4 이하일 때 사망 처리

ChoiHyebin opened the pull request:
✨ 포만감 10초과, 포만감 4 이하일 때 다마고치 사망 처리

ChoiHyebin committed:
feat : 포만감 10 초과, 포만감 4 이하일 때 사망 처리

dongkyeomjang committed:
Merge branch 'main' into main

dongkyeomjang committed:
Update Drawable.java

dongkyeomjang commented on src/MyFrame.java:
쪼금 이상한 부분들에 대해서 코멘틀 달아놓을게요. 확인해주시고 회의때 말씀해주세요!

dongkyeomjang commented on src/MyFrame.java:
```diff
@@ -140,16 +144,28 @@ else if(e.getSource() == cleanButton){
+		Random rand = new Random();
         // hungryTimer 생성 및 시작
-        hungryTimer = new Timer(10000, e -> {
+        hungryTimer = new Timer(2000, e -> {
```
타이머를 2000으로 바꾼건, 제 생각엔 테스트 때문인 것 같은데 PR을 보낼 땐 테스트용 코드로 보내면 안되고 바꿔서 보내야합니다.
기존의 10000으로 수정해주세요!

dongkyeomjang commented on src/MyFrame.java:
```diff
@@ -140,16 +144,28 @@ else if(e.getSource() == cleanButton){
+        
         // sleepyTimer 생성 및 시작
-        sleepyTimer = new Timer(20000, e -> {
+        sleepyTimer = new Timer(5000, e -> {
```
얘도 마찬가지로 타이머 수정 필요

dongkyeomjang commented on src/TamaManager.java:
```diff
@@ -1,3 +1,26 @@
+//import java.util.ArrayList;
```
필요없는 주석은 지워주세요

dongkyeomjang commented on src/TamaManager.java:
```diff
@@ -7,31 +30,50 @@ public class TamaManager {
-    public TamaManager(){
+    
+    Random rand = new Random();
+    private int food;
```
저는 food가 왜 멤버변수로 있어야하는지를 잘 모르겠어요. 코드를 읽어보니, food가 이제 밥 버튼을 눌렀을 때 1~3으로 랜덤하게 올릴 포만감 값을 의미하던데, 그냥 feed() 메소드 내부에서 따로 변수로 선언할 필요 없이 처리해도 무방하지 않나 싶네요
다른 생각 있으셨다면 회의 때 말씀해주세요!

ChoiHyebin replied:
food를 선언한 이유는 밥을 먹일 때 1에서 3 사이의 숫자로 게이지가 차게 되는데 이때 게이지가 11 이상이거나 15가 되었을 때 죽게 되잖아요. 아직 GUI가 제대로 완성된 상태가 아니라 포만감 게이지가 언제부터 정확히 11 이상인지 확인을 할 수가 없어 게이지가 11 이상이거나 15가 되었을 때 죽는 것을 콘솔 창에서 확인하기 위해 MyFrame에서 TamaManager의 getFood() 메소드를 통해 밥이 얼마나 먹여졌는지 호출하느라 food 변수를 선언했었어요. 확인 작업에서 사용했던 거라 최종 코드에선 지워도 괜찮습니다!

dongkyeomjang commented on src/TamaManager.java:
```diff
@@ -7,31 +30,50 @@ public class TamaManager {
+    
+    Random rand = new Random();
+    private int food;
+    private int state;
```
state도 필요없는 멤버변수라 생각하는데 이건 의견이 좀 갈릴 수 있을것같아서,, 근데 있더라도 TamaMangeer가 아니라 Tamagochi가 갖고있는게 좀 더 맞는것같습니다.
저는 tama 레퍼런스와 연결된 값이 있으면 살아있는 것, tama 레퍼런스의 값이 null이라면 죽은 것으로 상태를 구분하려고 했거든요

dongkyeomjang commented on src/TamaManager.java:
```diff
@@ -7,31 +30,50 @@ public class TamaManager {
         tama = new Tamagochi(x, y, size, "src/img/tamagochiImg.png", nickname);
     }
+    
+    public void repaintTama() {
```
얘는 왜 있는지 모르겠어요. 그냥 다시 createTama() 불러오면 되는거 아닌가요...?

ChoiHyebin replied:
네.. 그래서 안 쓰고 냅뒀는데 지워야 됩니다..

dongkyeomjang commented on src/TamaManager.java:
```diff
@@ -7,31 +30,50 @@ public class TamaManager {
+        // 포만감이 11 이상인 경우, 일정 확률로 Tamagochi의 dieByEating() 메소드 호출 => 배부른 상태에서 밥 먹이면 50% 확률로 죽음
+        if(tama.getSatiety() >= 11) {
+        	if(Math.random() < 0.5) {
+        		state = 0;
```
만약 tama 의 레퍼런스 null 체크로 죽었는지 살았는지 판단하는거라면, "if (state==0)" 이 부분이 if (tama==null) 요런식으로 바뀔 수 있겠죠?

dongkyeomjang commented on src/Tamagochi.java:
```diff
@@ -1,3 +1,47 @@
+//import java.awt.Graphics;
```
필요없는 주석 제거

dongkyeomjang commented on src/Tamagochi.java:
```diff
@@ -26,11 +70,19 @@ public void display(Graphics g){
+//        return new Tombstone(getX(), getY(), getSize(), "src/img/tombstoneImg1.png",causeOfDeath);
+//    }
+    
+    public Tombstone dieByEating(String causeOfDeath) {
```
얘는 if문이 필요 없어요. 왜냐면 매니저에서 해당 로직은 전부 처리된상태로 dieByEating을 호출할거거든요.

dongkyeomjang replied:
저부분 수정해야합니다.

ChoiHyebin replied:
아하 알겠습니다

dongkyeomjang reviewed (changes requested):
수정이 필요합니다

ChoiHyebin committed:
Update MyFrame.java

✨ 다마고치의 캐릭터가 사망했을 때 타이머를 stop() 해주기 위해 수정

ChoiHyebin committed:
Update MyFrame.java

✨ 다마고치의 캐릭터가 사망했을 때 타이머를 stop() 하기 위해 수정

ChoiHyebin committed:
Update TamaManager.java

✨ 다마고치의 캐릭터가 사망했을 때 타이머를 멈추게 하기 위해 feed() 메소드와 gettingHungry() 메소드가 hungryTimer와 sleepyTimer를 매개변수로 받아 타이머를 멈추게 함. 캐릭터가 사망했을 때 사망 알림창이 뜸.

ChoiHyebin committed:
Update Tamagochi.java
